### ISSUE TITLE
Internal code review

### ISSUE THREAD

**Issue by mariocao** [witnet/witnet-ethereum-bridge#147]
The goal of this issue is to perform a similar code review as done in #68.

**Comment by mariocao** [comment]
## Code review result

Tool used: [SmartDec](https://tool.smartdec.net/)
Commit: https://github.com/witnet/witnet-ethereum-bridge/commit/3cec92f8a7b86bc1a0957b402eb5bdcf969f0444
Resultt: [SmartDec Scan](https://tool.smartdec.net/scan/7f1f7b5ad4ce4707b40533cf4621d25b)

### Summary

```
·----------------------------------------------·--------------·
|                Error Type                    |  Occurences  |
·······························································
| Hardcoded address                            |           1  |
·······························································
| Multiplication after division                |           1  |
·······························································
| Costly loop                                  |           5  | 
·······························································
| Locked money                                 |           5  |
·······························································
| Compiler version not fixed                   |           7  |
·······························································
| Pure-functions should not read/change state  |           9  |
·······························································
| Upgrade code to Solidity 0.5.x               |           1  |
·······························································
| Prefer external to public visibility level   |          27  | 
·······························································
| Use of assembly                              |          10  |
·······························································
| Implicit visibility level                    |           2  |
·----------------------------------------------·--------------·
| TOTAL                                        |          68  |
·----------------------------------------------·--------------·
```

**Comment by mariocao** [comment]
Closed in #148
